packages/back-end/src/integrations/SqlIntegration.ts

```diff
@@ -3065,55 +3129,86 @@ AND event_name = '${eventName}'`,
       (metric) => metric.sql === sqlQuery && metric.type === type
     );
   }
+  getAutoGeneratedFactTableSqlQuery(
+    eventName: string,
+    schemaFormat: AutoFactTableSchemas,
+    schema?: string
+  ): string {
+    const {
+      timestampColumn,
+      userIdColumn,
+      anonymousIdColumn,
+      getTrackedEventTablePath,
+      getEventFilterWhereClause,
+      filterColumns,
+      getDateLimitClause,
+    } = this.getSchemaFormatConfig(schemaFormat);
+
+    const sqlQuery = `
+      SELECT
+        ${userIdColumn} as user_id,
```

Comment:
Didn't we have a `hasUserId` setting before?  I thought Rudderstack did not always add this column

Reply:
You're correct - I misread Rudderstack's documentation the first time. I've updated this to use the same logic as Auto Metrics does for the `user_id` column.